### DISCUSSION TITLE
Make it simple and direct to set icon like so

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#8515] [Ideas]
This would much simplify setting a tiddlers *icon*:

Currently, the `icon` field expects a *tiddler title* because this is then *transcluded* to show the content. It would be a lot simpler if the user could just *directly* add anything in the icon field (e.g an emoji symbol) and have this show as an icon - i.e IF it is a tiddler title then it is, still, transcluded, but anything else is shown as-is.

Currently [$:/core/ui/TiddlerIcon](https://tiddlywiki.com/#%24%3A%2Fcore%2Fui%2FTiddlerIcon) reads:

```
\whitespace trim
<$let tiddlerIcon={{{ [<currentTiddler>] :cascade[all[shadows+tiddlers]tag[$:/tags/TiddlerIconFilter]!is[draft]get[text]] }}}>
<$list filter="[<tiddlerIcon>!is[blank]]" variable="ignore">
<$transclude tiddler=<<tiddlerIcon>>/>
</$list>
</$let>
```

I would like for `!is[blank]` it to instead be `has[title]` and add `emptyMessage=<<tiddlerIcon>>`:

```
\whitespace trim
<$let tiddlerIcon={{{ [<currentTiddler>] :cascade[all[shadows+tiddlers]tag[$:/tags/TiddlerIconFilter]!is[draft]get[text]] }}}>
<$list filter="[<tiddlerIcon>has[title]]" variable="ignore" emptyMessage=<<tiddlerIcon>> >
<$transclude tiddler=<<tiddlerIcon>>/>
</$list>
</$let>
```
(`has[title]` captures both shadows and non-shadows)

What you think about this for core?

**Comment by Leilei332** [comment]
I think we can let the tiddler use the text in the `icon` field as icon when the field only has one character (and the tiddler which has the same title doesn't exist).

I think it is more useful to create a new icon cascade for it, which is more compat than the original one.

Currently the tiddlywiki edition which is published by [TidGi](https://github.com/tiddly-gittly/TidGi-Desktop) supports using emoji as tiddler icon.

**Reply by twMat** [reply]
> when the field only has one character

Ah, that's a neat idea!

> new icon cascade for it, which is more compat than the original one.

How would that be?

**Reply by Leilei332** [reply]
I think a new icon cascade can be created which works like the color cascade, like this (apply 💡 as icon to tiddler tagged `idea`):

```
[tag[idea]then[💡]]
```
